File: library/src/main/java/com/gucci/library/ILiveBus.java
package com.gucci.library;

import android.arch.lifecycle.Lifecycle;

/**
 * Created by 李志云 2019/3/6 01:30
 */
public interface ILiveBus {
    void sendEvent(int key, Object value);
    void sendStickEvent(int key, Object value);
    void registerStick(LiveBus.EventObserver eventObserver);



    void listener(Lifecycle lifecycle, LiveBus.EventObserver eventObserver);
    void register(LiveBus.EventObserver eventObserver);
    void unregister(LiveBus.EventObserver eventObserver);
}

File: library/src/main/java/com/gucci/library/ILiveBus.kt
package com.gucci.library

import com.gucci.lifecycle.LifecycleListener


interface ILiveBus {
    fun sendEvent(key: Int, value: Any)
    fun sendStickEvent(key: Int, value: Any)
    fun registerStick(eventObserver: EventObserver<*>)


    fun listener(eventObserver: EventObserver<*>):LifecycleListener
    fun register(eventObserver: EventObserver<*>)
    fun unregister(eventObserver: EventObserver<*>)
}
